### PR TITLE
Cfkey trust key redmine3074

### PR DESCRIPTION
**Pull request by jimis**
Replaces pull request #1223.

### CODE REVIEW

**Comment by jimis** [comment]
@riccardomurri this pull request moves your fixes to `--trust-key` option, and my testing reveals that it works both hub-side and client side. I also updated some stylistic issues and added some log messages. Let me know what you think and if you are able to test.

BTW the commit dates are crazy old, it's impossible to follow `git log`.

**Comment by jimis** [comment]
trigger build

File: cf-key/cf-key-functions.c [hunk 1]
```diff
@@ -117,23 +129,98 @@ int PrintDigest(const char* pubkey)
...
+    if (ipaddress != NULL)
+    {
+        Log(LOG_LEVEL_VERBOSE,
+            "Adding a CONNECT entry in lastseen db: IP '%s', key '%s'",
+            ipaddress, digest);
+        LastSaw1(ipaddress, digest, LAST_SEEN_ROLE_CONNECT);
+    }
 
-    ok = CopyRegularFileDisk(pubkey, outfilename);
+    SavePublicKey(username, digest, key);
```
Comment (riccardomurri) [inline cf-key/cf-key-functions.c]: If the save operation fails (e.g., read-only filesystem), the function still exists with code 0 (== OK). Can you correct this?

**Comment by jimis** [comment]
Updated to check return value of `SavePublicKey()`. Unfortunately the rest of the callers of `SavePublicKey()` still don't check the return value.

**Comment by jimis** [comment]
I'll merge later today since the test build seems successful or maybe @kacfengine you want to review and merge. 

@riccardomurri thanks for bringing the initial pull request, and sorry it took almost two years! :-)

**Comment by kacfengine** [comment]
No detailed code review from me, this is your area. But looks good overall!

**Pull request merged**